src/library_browser.js

```diff
@@ -977,15 +977,29 @@ var LibraryBrowser = {
         if (onload) {{{ makeDynCall('viiii', 'onload') }}}(handle, arg, buffer, byteArray.length);
         if (free) _free(buffer);
       } else {
-        if (onerror) {{{ makeDynCall('viiii', 'onerror') }}}(handle, arg, http.status, http.statusText);
+        if (onerror) {
+          var statusText = 0;
+          if(http.statusText) {
+            var len = lengthBytesUTF8(http.statusText) + 1;
+            statusText = stackAlloc(len);
+            stringToUTF8(http.statusText, statusText, len);
```

Comment:
@kripken do we have any shorthand for this?  Seems like it would be a fairly common pattern.